### ISSUE TITLE
Ship crashes: out of loom

### ISSUE THREAD

**Issue by codydh** [urbit/urbit#3740]
**Describe the bug**
Ship crashes with error:
```allocate: reclaim: half of 3 entries

bail: meme
bailing out
pier: serf error: end of file
dojo> 
address 0x438 out of loom!
loom: [0x200000000 : 0x280000000)
Assertion '0' failed in noun/events.c:129

bail: oops
bailing out
Assertion failed: 0 (noun/manage.c: u3m_bail: 677)
Aborted (core dumped)
```

I followed all steps here: https://github.com/urbit/urbit/issues/3645#issuecomment-709631355

My order of steps was:
- Booting without network access to run `|pack`
- Using the process to trigger `|trim`
- Using the alternate binary to `|meld`

This got me up and running for about 6–7 hours, after which I crashed with the same error.

**System (please supply the following information, if relevant):**
- OS: linux64 and macOS 10.15.7
- [%base-hash ~[0v17.5ufk3.g8q69.u65fr.cmtgk.vu62l.rhhkh.serne.gee8g.e77p6.ht5ec]

**Additional context**
My pier is 6.2GB. This began very shortly after applying the last OTA.
```
|mass
%arvo:   
  %hoon: MB/4.398.632   
  %zuse: MB/2.950.096 
  %vane:       
    %reports:    
      %vane-a:          
        %peers-known: MB/17.147.952
        %peers-alien: MB/1.024.844
      --MB/18.172.796
      %vane-b: 
        %timers: MB/2.721.108
      --MB/2.721.108      
      %vane-c:        
        %domestic:  
          %base:    
            %ankh: MB/29.202.860
            %mime: MB/6.820.344
            %ford-vases: B/0
            %ford-marks: B/0
            %ford-casts: B/0
          --MB/36.023.204  
          %home:        
            %ankh: MB/2.364.008
            %mime: KB/96.636
            %ford-vases: MB/24.842.160
            %ford-marks: MB/114.290.836
            %ford-casts: B/0
          --MB/141.593.640 
          %kids:    
            %ankh: B/576 
            %mime: KB/1.080
            %ford-vases: B/216
            %ford-marks: KB/12.912
            %ford-casts: KB/1.968
          --KB/16.752
        --MB/177.633.596
        %foreign: MB/12.934.700
        %object-store: 
          %commits: KB/587.556
          %blobs: B/0          
        --KB/587.556
      --MB/191.155.852      
      %vane-d:             
        %hey: B/24              
        %dug: KB/1.632
      --KB/1.656           
      %vane-e:          
        %bindings: KB/3.964
        %auth: KB/64.428       
        %connections: B/0        
        %channels: MB/593.120.756
        %axle: KB/9.692       
      --MB/593.198.840
      %vane-f:            
      --B/0            
      %vane-g:         
        %foreign: KB/77.356   
        %blocked:         
        --B/0          
        %active: 
          %acme: KB/52.916
          %azimuth-tracker: KB/1.556
          %chat-cli: MB/50.063.436
          %chat-hook: KB/385.616
          %chat-store: MB/82.653.008 
          %chat-view: KB/366.280  
          %clock: B/984
          %contact-hook: KB/147.776
          %contact-store: KB/79.048                                                                                                                             
          %contact-view: KB/408.936                                             
          %dojo: KB/8.444                                                       
          %eth-watcher: MB/5.235.656                                            
          %file-server: MB/12.716.056                                           
          %glob: KB/1.164
          %goad: KB/1.044
          %graph-pull-hook: KB/1.032
          %graph-push-hook: KB/1.152
          %graph-store: KB/541.284
          %group-hook: KB/1.440
          %group-pull-hook: KB/4.596                                            
          %group-push-hook: KB/1.368
          %group-store: KB/456.448                                              
          %hood: KB/48.408                                                                                                                                      
          %invite-hook: B/960
          %invite-store: KB/274.532
          %invite-view: KB/253.888     
          %launch: KB/380.672
          %lens: B/816     
          %link-listen-hook: KB/13.376
          %link-proxy-hook: KB/6.008
          %link-store: KB/10.800
          %link-view: KB/45.492
          %metadata-hook: KB/2.796
          %metadata-store: KB/570.624
          %permission-group-hook: B/720
          %permission-hook: KB/2.304                                            
          %permission-store: KB/90.844                                          
          %ping: KB/1.416                                                       
          %publish: KB/477.692
          %s3-store: KB/271.368
          %soto: B/720
          %spider: KB/2.600
          %weather: KB/314.404
        --MB/155.899.680
      --MB/155.977.036
      %vane-i: 
        %nex: B/0
        %outbound: B/160
        %by-id: B/0                
        %by-duct: B/0             
        %axle: B/132 
      --B/292  
      %vane-j:               
        %pki: MB/1.887.032
        %etn: KB/8.736
      --MB/1.895.768
    --MB/963.123.348
    %caches:                    
      %vane-a: MB/2.286.476    
      %vane-b: KB/325.488   
      %vane-c: KB/49.104    
      %vane-d: KB/441.920   
      %vane-e: MB/1.729.540
      %vane-f: KB/14.448
      %vane-g: KB/458.148      
      %vane-i: KB/186.188   
      %vane-j: KB/930.004             
    --MB/6.421.316                     
    %dregs:                 
      %vane-a: MB/1.045.080
      %vane-b: B/440
      %vane-c: KB/116.884
      %vane-d: B/288       
      %vane-e: B/192          
      %vane-f: B/120              
      %vane-g: MB/1.384.360      
      %vane-i: B/192 
      %vane-j: B/240    
    --MB/2.547.796             
  --MB/972.092.460     
--MB/979.441.188              
total userspace: MB/979.441.188
  kernel: KB/103.476
total arvo stuff: KB/103.476
  warm jet state: KB/92.988
  cold jet state: MB/155.078.300
  hank cache: B/372   
  battery hash cache: B/288
  call site cache: B/100
  hot jet state: KB/125.348
total jet stuff: MB/155.297.396
  bytecode programs: MB/1.289.196
  bytecode cache: KB/117.552     
total nock stuff: MB/1.406.748
  trace buffer: B/36  
  memoization cache: B/288
total road stuff: B/324
space profile: KB/8.868
total marked: GB/1.136.258.000
free lists: MB/116.184.444
sweep: GB/1.136.258.000
```

**Comment by joemfb** [comment]
@codydh, your hitting an issue with (the absence of ) backpressure in  %eyre (our http server vane), specifically with regards to its SSE channels. A fix is in development in #3680. In the meantime, you should have `|trim` as of the last OTA; running it and then `|pack` should get you back on your feet (although you may need to keep doing that periodically until the fix is released).

**Comment by codydh** [comment]
@joemfb I was able to run those to temporarily get running again, but it recurred so I wound up breaching. Thanks for the help!
